src/engine/controls/loopingcontrol.cpp

```diff
@@ -959,18 +959,17 @@ void LoopingControl::slotLoopStartPos(double positionSamples) {
     if (!position.isValid()) {
         emit loopReset();
         setLoopingEnabled(false);
+    } else if (loopInfo.endPosition.isValid() && loopInfo.endPosition <= position) {
+        emit loopReset();
+        loopInfo.endPosition = mixxx::audio::kInvalidFramePos;
+        m_pCOLoopEndPosition->set(kNoTrigger);
+        setLoopingEnabled(false);
     }
 
     loopInfo.seekMode = LoopSeekMode::MovedOut;
```

Comment:
Let's move these lines up, closer to where `loopInfo` is initialized.

Then use `loopInfo.startPosition` instead of `position` for consistency and to reveal the dependencies. For me this is easier to follow. Less variables to remember, more concise.

Compile time enforcement: Use a local scope so that the temporary `position` becomes unavailable for the remainder of the function body.

Reply:
https://github.com/mixxxdj/mixxx/pull/4266/commits/6fe90117fbd741517212eb080fe9829e6abe38df